app/src/main/java/com/gathering/android/event/myevent/MyEventScreen.kt

```diff
@@ -43,74 +52,110 @@ class MyEventScreen : Fragment(), MyEventNavigator {
         super.onViewCreated(view, savedInstanceState)
-        val linearLayoutManager =
-            LinearLayoutManager(requireContext(), LinearLayoutManager.VERTICAL, false)
-        binding.rvEvent.layoutManager = linearLayoutManager
-
-        swipeGesture = object : SwipeGesture(requireContext()) {
-            override fun onSwiped(viewHolder: RecyclerView.ViewHolder, direction: Int) {
-                val position = viewHolder.absoluteAdapterPosition
-                val event = adapter.getEventAtPosition(position)
-                event?.let {
-                    if (direction == ItemTouchHelper.LEFT) {
-                        viewModel.onSwipedToDelete(it)
-                        // Show a snack bar with an "Undo" option for the delete action
-                        val snackBar = Snackbar.make(
-                            binding.root, EVENT_DELETED, Snackbar.LENGTH_LONG
-                        )
-                        snackBar.setAction(UNDO) {
-                            viewModel.onUndoDeleteEvent()
+        if (isComposeEnabled) {
+            viewModel.onViewCreated(this)
+            return
+        } else {
+            val linearLayoutManager =
+                LinearLayoutManager(requireContext(), LinearLayoutManager.VERTICAL, false)
+            binding.rvEvent.layoutManager = linearLayoutManager
+
+            swipeGesture = object : SwipeGesture(requireContext()) {
+                override fun onSwiped(viewHolder: RecyclerView.ViewHolder, direction: Int) {
+                    val position = viewHolder.absoluteAdapterPosition
+                    val event = adapter.getEventAtPosition(position)
+                    event?.let {
+                        if (direction == ItemTouchHelper.LEFT) {
+                            viewModel.onSwipedToDelete(it)
+                            // Show a snack bar with an "Undo" option for the delete action
+                            val snackBar = Snackbar.make(
+                                binding.root, EVENT_DELETED, Snackbar.LENGTH_LONG
+                            )
+                            snackBar.setAction(UNDO) {
+                                viewModel.onUndoDeleteEvent(event)
+                            }
+                            snackBar.show()
                         }
-                        snackBar.show()
                     }
                 }
             }
-        }
 
-        itemTouchHelper = ItemTouchHelper(swipeGesture)
+            itemTouchHelper = ItemTouchHelper(swipeGesture)
 
-        binding.rvEvent.adapter = adapter
-        binding.rvEvent.addOnScrollListener(EndlessScrollListener {
-            viewModel.onNextPageRequested()
-        })
-        itemTouchHelper.attachToRecyclerView(binding.rvEvent)
+            binding.rvEvent.adapter = adapter
+            binding.rvEvent.addOnScrollListener(EndlessScrollListener {
+                viewModel.onNextPageRequested()
```

Comment:
i dont see this method called from compose view anywhere !! how do we handle next pages in compose ?

Reply:
i have created a ticket for this. since this is similar in HomeScreen. i have to handle this in both screens. 
https://trello.com/c/TFU72EJu/151-handle-onnextpagerequested-in-homescreen-and-myeventscreen